pallets/subtensor/src/coinbase/run_coinbase.rs

```diff
@@ -199,59 +199,73 @@ impl<T: Config> Pallet<T> {
-            .to_num::<u64>();
-        // NOTE: Only the validation emission should be split amongst parents.
-
-        // --- 2. Compute the remaining emission after the hotkey's share is deducted.
-        let emission_minus_take: u64 = validating_emission.saturating_sub(hotkey_take);
+        let childkey_take_proportion: I64F64 =
+            I64F64::from_num(Self::get_childkey_take(hotkey, netuid))
+                .saturating_div(I64F64::from_num(u16::MAX));
+        let mut total_childkey_take: u64 = 0;
 
-        // --- 3. Track the remaining emission for accounting purposes.
-        let mut remaining_emission: u64 = emission_minus_take;
+        // --- 2. Track the remaining emission for accounting purposes.
+        let mut remaining_emission: u64 = validating_emission;
 
-        // --- 4. Calculate the total stake of the hotkey, adjusted by the stakes of parents and children.
+        // --- 3. Calculate the total stake of the hotkey, adjusted by the stakes of parents and children.
         // Parents contribute to the stake, while children reduce it.
         // If this value is zero, no distribution to anyone is necessary.
         let total_hotkey_stake: u64 = Self::get_stake_for_hotkey_on_subnet(hotkey, netuid);
         if total_hotkey_stake != 0 {
-            // --- 5. If the total stake is not zero, iterate over each parent to determine their contribution to the hotkey's stake,
+            // --- 4. If the total stake is not zero, iterate over each parent to determine their contribution to the hotkey's stake,
             // and calculate their share of the emission accordingly.
             for (proportion, parent) in Self::get_parents(hotkey, netuid) {
-                // --- 5.1 Retrieve the parent's stake. This is the raw stake value including nominators.
+                // --- 4.1 Retrieve the parent's stake. This is the raw stake value including nominators.
                 let parent_stake: u64 = Self::get_total_stake_for_hotkey(&parent);
 
-                // --- 5.2 Calculate the portion of the hotkey's total stake contributed by this parent.
+                // --- 4.2 Calculate the portion of the hotkey's total stake contributed by this parent.
                 // Then, determine the parent's share of the remaining emission.
                 let stake_from_parent: I96F32 = I96F32::from_num(parent_stake).saturating_mul(
                     I96F32::from_num(proportion).saturating_div(I96F32::from_num(u64::MAX)),
                 );
                 let proportion_from_parent: I96F32 =
                     stake_from_parent.saturating_div(I96F32::from_num(total_hotkey_stake));
-                let parent_emission_take: u64 = proportion_from_parent
-                    .saturating_mul(I96F32::from_num(emission_minus_take))
+                let parent_emission: u64 = proportion_from_parent
+                    .saturating_mul(I96F32::from_num(validating_emission))
+                    .to_num::<u64>();
```

Comment:
probably shouldn't cast here (keep as a float). Maybe only cast the final results.